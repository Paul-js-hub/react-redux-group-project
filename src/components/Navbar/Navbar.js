/* eslint-disable import/no-named-default */
import { NavLink } from 'react-router-dom';
import { default as BootstrapNavbar } from 'react-bootstrap/Navbar';
import Container from 'react-bootstrap/Container';
import logo from '../../img/logo.png';
import style from './navbar.module.css';

const Navbar = () => (
  <BootstrapNavbar fixed="top" bg="white" variant="light" className={style.navContainer} expand="lg">
    <Container fluid>
      <BootstrapNavbar.Brand href="#home" className="me-auto">
        <img
          alt=""
          src={logo}
          width="30"
          height="30"
          className="d-inline-block align-top"
        />
        <h3 className="d-inline">React Bootstrap</h3>
      </BootstrapNavbar.Brand>
      <BootstrapNavbar.Toggle aria-controls="basic-navbar-nav" />
      <BootstrapNavbar.Collapse className={style.linkContainer}>
        <NavLink className={({ isActive }) => (isActive ? style.active : '')} to="/">Rockets</NavLink>
        <NavLink className={({ isActive }) => (isActive ? style.active : '')} to="/missions">Missions</NavLink>
        <div className="vr mx-3" />
        <NavLink className={({ isActive }) => (isActive ? style.active : '')} to="/profile">My Profile</NavLink>
      </BootstrapNavbar.Collapse>
    </Container>
  </BootstrapNavbar>
);

export default Navbar;
