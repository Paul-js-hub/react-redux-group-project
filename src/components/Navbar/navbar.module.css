.navContainer {
  border-bottom: 1px solid #ccc;
}

.navContainer a {
  text-decoration: none;
  margin-right: 1rem;
  display: block;
  margin-top: 10px;
  margin-bottom: 10px;
}

.navContainer a.active {
  text-decoration: underline;
}

.navContainer img {
  margin-right: 10px;
}

.linkContainer {
  max-width: 300px;
}
