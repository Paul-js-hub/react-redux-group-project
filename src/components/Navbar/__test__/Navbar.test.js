import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Test of navbar', () => {
  it('test if the navbar is rendered', () => {
    const navbar = renderer
      .create(<Router><Navbar /></Router>)
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
