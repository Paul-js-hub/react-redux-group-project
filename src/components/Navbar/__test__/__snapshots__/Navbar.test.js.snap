// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of navbar test if the navbar is rendered 1`] = `
<nav
  className="navContainer navbar navbar-expand-lg navbar-light bg-white fixed-top"
>
  <div
    className="container"
  >
    <div>
      <a
        className="me-auto navbar-brand"
        href="/"
      >
        <img
          alt=""
          className="d-inline-block align-top"
          height="30"
          src="logo.png"
          width="30"
        />
        <h3
          className="d-inline"
        >
          SpaceX
        </h3>
      </a>
    </div>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="linkContainer navbar-collapse collapse"
    >
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className=""
        href="#/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <div
        className="vr mx-3"
      />
      <a
        className=""
        href="#/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </div>
  </div>
</nav>
`;
